### ISSUE TITLE
Port WebJobs SDK to .NET Core/ ASP.NET 5 Core

### ISSUE THREAD

**Issue by rustd** [Azure/azure-webjobs-sdk#541]

**Comment by jefking** [comment]
I would love to help out with this.

**Comment by turowicz** [comment]
How about this?

https://blog.kloud.com.au/2016/06/08/azure-webjobs-with-dotnet-core-rc2/

**Comment by armartinez** [comment]
That example is only for deploying a dotnet core application to an Azure WebJob, but the SDK has built-in features for working with Azure Storage and Service Bus, for scheduling tasks and handling errors and more which are pretty useful. I'm using hangfire running on a webjob since it supports dotnet core and has many of the features of the SDK, still I would preffer just using the SDK.

**Comment by mb2140** [comment]
Is this being planned for the 2.0 release?  I just grabbed the preview version and it doesn't look like it's supported yet.

**Comment by oliman** [comment]
I'm also interested in any updates to using the WebJobs SDK in a dot net core webjob either officially or if there is any homegrown solutions while we wait.

**Comment by damienpontifex** [comment]
As @jefking had said, I would also love to be able to contribute to this effort. 

Some dependencies (i.e. Azure Storage) now have .net core support so maybe a plan to enable some triggers/features at first. Would there be a checklist/plan that we could make in an effort to enable beginning work on this?

**Comment by fabiocav** [comment]
We have an issue tracking the work that will need to happen across the different repos (SDK, extensions and Functions Runtime) in the meta repo: https://github.com/Azure/Azure-Functions/issues/98

We'll be creating the more detailed issues in each specific repo as planning moves forward.

The planning activities around this effort have been assigned to our next milestone, starting in January, and we should start executing on those plans once that is done (we'll have more details once we scope this out)

**Comment by imsam67** [comment]
Any updates on WebJobs SDK in .NET Core?

**Comment by fabiocav** [comment]
@imsam67 yeah, there has been some progress on this work now that //build is out of the way. We'll be updating the issue tracking the work with the details and a migration branch and PR going in soon.

**Comment by imsam67** [comment]
@fabiocav thank you!

**Comment by wind-rider** [comment]
I guess work on this was (partly?) done in https://github.com/Azure/azure-webjobs-sdk/pull/1193

**Comment by fabiocav** [comment]
The core work has been done and merged into the `dev` branch. The remaining (mostly engineering) tasks will be completed as migration work progresses on the extensions and script repositories.

Beta packages targeting .NET Standard 2.0 have been released to NuGet.

**Comment by imsam67** [comment]
@fabiocav Is this the NuGet package you were referring to? Thanks.
https://www.nuget.org/packages/Microsoft.Azure.WebJobs.Core/2.0.0-beta1

**Comment by johnkwaters** [comment]
I made some good progress on publishing a .netcore webjob, but am currently stuck getting it to run in my .netcore azure app service. I have the binaries in the app_data/jobs/continuous folder, and a cmd file that does "dotnet myassemblyname.dll", and i see it gets called. But in the webjob log I see "'﻿dotnet' is not recognized as an internal or external command". Any pointers?

**Comment by imsam67** [comment]
Two points: in my case, I have my web job in its own folder under `app_data/jobs/continuous` so it's `app_data/jobs/continuous/mywebjob`. Second, I had to do a self contained deployment with runtimes and all dependent packages in the same folder. This approach made all my headaches go away. You can do `dotnet publish --self-contained -r win-x64 -c Release` and FTP the contents of the publish folder into the folder where the web jobs will run.

**Comment by johnkwaters** [comment]
When I do that, I get a myassemblyname.exe and still the error dotnet not found... should I then remove the run.cmd and just execute the exe without a dotnet command?
